bin/hipfc

```diff
@@ -252,11 +252,12 @@ cdir=$(getdname $0)
 HOW_CALLED=${0##*/}
 
 CUDA_HOME=${CUDA_HOME:-/usr/local/cuda}
-ROCM_DIR=${ROCM_DIR:-/opt/rocm}
+ROCM_PATH=${ROCM_PATH:-/opt/rocm}
+HIP_PLATFORM=${HIP_PLATFORM:-amd}
```

Comment:
Until the next official ROCm version that works with `amd` instead of `hcc` is released, I would
prefer if we could keep the default as `hcc`.

Reply:
Is that what happened in HIP 4.0.0 ? I had some builds break this week on a fresh install with HIP on AMD platforms; I saw errors that indicated hcc was not a recognized platform. Do we want to bring this into the rocm-4.0.x branch rather than develop ?

Reply:
Hello txomin13, ROCm4.0 is already released in last December.
We need to keep default as "amd". We have the issue from ROCm4.0 itself.

schoonovernumerics, he has also confirm the breakage due to "hcc"
We need to patch it in rocm-4.0.x too along with master.

Reply:
@pramenku Okay, wasn't aware that change happened already with 4.0.0. Then, we keep it as you proposed.

Reply:
This sounds like a good plan. My current builds based on 4.0.0 have switched to using gfortran for the fortran compiler and hipcc for compiling HIP kernels. This will get projects back to using hipfc.

Reply:
Thanks all for the input.
Dominic, can we merge it now.
Please make sure to merge into "master" and "rocm-4.0.x".

Reply:
@pramenku See my review below.